src/iarc7_motion/iarc_tasks/hit_roomba_task.py

```diff
@@ -0,0 +1,214 @@
+            self._MAX_Z_VELOCITY = rospy.get_param('~max_z_velocity')
+            self._K_X = rospy.get_param('~k_term_tracking_x')
+            self._K_Y = rospy.get_param('~k_term_tracking_y')
+            self._descent_velocity = rospy.get_param('~hit_descent_velocity')
+        except KeyError as e:
+            rospy.logerr('Could not lookup a parameter for hit roomba task')
+            raise
+
+        self._state = HitRoombaTaskState.init
+
+    def _receive_roomba_status(self, data):
+        with self._lock:
+            self._roomba_array = data
+
+    def _receive_switch_status(self, data):
+        with self._lock:
+            self._switch_message = data
+
+    def _current_velocity_callback(self, data):
+        with self._lock:
+            self._drone_odometry = data
+
+    def get_desired_command(self):
+        with self._lock:
+            if self._canceled:
+                return (TaskCanceled(),)
+
+            elif (self._state == HitRoombaTaskState.init):
+                if self._roomba_array is None or self._drone_odometry is None:
+                    self._state = HitRoombaTaskState.waiting
+                else:
+                    self._state = HitRoombaTaskState.descent
+                return (TaskRunning(), NopCommand())
+
+            elif (self._state == HitRoombaTaskState.waiting):
+                if self._roomba_array is None or self._drone_odometry is None:
+                    self._state = HitRoombaTaskState.waiting
+                else:
+                    self._state = HitRoombaTaskState.descent
+                return (TaskRunning(), NopCommand())
+
+            elif (self._state == HitRoombaTaskState.descent):
+                try:
+                    roomba_transform = self._tf_buffer.lookup_transform(
+                                        'level_quad',
+                                        self._roomba_id,
+                                        rospy.Time.now(),
+                                        rospy.Duration(self._TRANSFORM_TIMEOUT))
+                except (tf2_ros.LookupException,
+                        tf2_ros.ConnectivityException,
+                        tf2_ros.ExtrapolationException) as ex:
+                    rospy.logerr('ObjectTrackTask: Exception when looking up transform')
+                    rospy.logerr(ex.message)
+                    return (TaskAborted(msg='Exception when looking up transform during hit roomba'),)
+
+                # Creat point centered at drone's center
+                stamped_point = PointStamped()
+                stamped_point.point.x = 0
+                stamped_point.point.y = 0
+                stamped_point.point.z = 0
+
+                # returns point distances of roomba to center point of level quad
+                self._roomba_point = tf2_geometry_msgs.do_transform_point(
+                                                    stamped_point, roomba_transform)
+
+                if not self._check_max_roomba_range():
```

Comment:
Could you move this conditional and the next one above line 120?  Same reason as in `track_roomba_task.py`